### PR TITLE
Parser sockstat

### PR DESCRIPTION
**Pull request by vishwanathjadhav**
A parser is added to parse the contents of `/proc/net/sockstat`

### CODE REVIEW

File: insights/parsers/sockstat.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+"""
+
+
+from .. import Parser, parser, LegacyItemAccess
+from insights.parsers import SkipException
+from insights.specs import Specs
+
+
+@parser(Specs.sockstat)
+class SockStats(Parser, LegacyItemAccess):
```
Comment (xiangce) [inline insights/parsers/sockstat.py]: The `LegacyItemAccess` only works with `self.data` which is not in your class. In fact, the `LegacyItemAccess` will be deprecated soon. If you want the Parser can be accessed like a `dict`, you can mix it with `dict` directly.

FYI: https://github.com/RedHatInsights/insights-core/blob/89828b760399231917414c8326515fc97d79118c/insights/parsers/modinfo.py#L145

File: insights/parsers/sockstat.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
+"""
+SockStats - file ``/proc/net/sockstat``
+=======================================
+The ``TcpIpStats`` class implements the parsing of ``/proc/net/snmp``
+file, which contains TCP/IP stats of individual layer.
+"""
+
+
+from .. import Parser, parser, LegacyItemAccess
```
Comment (xiangce) [inline insights/parsers/sockstat.py]: Please use a uniform format for the `importing`s in one python module. E.g. all in relative path or all in full path

File: insights/parsers/sockstat.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                    'memory': '0'
+                  }
+        }
+
+    Raises:
+        SkipException: When contents are empty
+    """
+
+    def parse_content(self, content):
+        self.sock_stats = {}
```
Comment (xiangce) [inline insights/parsers/sockstat.py]: By `mixing` this parser with `dict` this attribute can be removed. If you want to keep it as an attribute, you could make it as a member-function decorated with `@property` and `return self` directly. And you should add it to the `Attributes` section of the docsting in that case as well.

File: insights/parsers/sockstat.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+        if not content:
+            raise SkipException("No Contents")
+
+        for line in content:
+            line_split = line.split(":")
+            nt_seg = line_split[0].lower()
+            self.sock_stats[nt_seg] = {}
+            val_lst = line_split[1].strip().split(' ')
```
Comment (xiangce) [inline insights/parsers/sockstat.py]: do you think the `split()` is  better than`split(' ')`?

Comment (vishwanathjadhav) [reply insights/parsers/sockstat.py]: Changed

File: insights/parsers/sockstat.py (hunk 3 above)
Comment (vishwanathjadhav) [reply insights/parsers/sockstat.py]: Done

File: insights/parsers/sockstat.py (hunk 2 above)
Comment (vishwanathjadhav) [reply insights/parsers/sockstat.py]: Done

File: insights/parsers/sockstat.py (hunk 1 above)
Comment (vishwanathjadhav) [reply insights/parsers/sockstat.py]: Done

**Comment by vishwanathjadhav** [comment]
Rule dependency:
https://gitlab.cee.redhat.com/insights-rules/insights-plugins/merge_requests/1977

**Pull request merged**